File: codes/array/283.move-zeroes/move-zeroes.cpp
#include <vector>
using namespace std;
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        if(nums.size()<=1){
            return;
        }
        auto i = nums.begin();
        auto j = nums.begin();
        while (j!=nums.end()){
            if(*j==0){
                j++;
            }else{
                *i++ = *j++;
                // 等价写法
                // *i=*j;
                // i++;
                // j++;
            }
        }
        while(i!=nums.end()){
            *i++ = 0;
        }
    }
};

File: codes/array/283.move-zeroes/move-zeroes.go
package main
func moveZeroes(nums []int)  {
	if len(nums)<=1{
		return
	}
	i:=0
	for j:=0;j<len(nums);{
		if nums[j]==0{
			j++
		}else{
			nums[i]=nums[j]
			i++
			j++
		}
	}
	for ;i<len(nums);i++{
		nums[i]=0
	}
}